pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -206,18 +205,23 @@ async def run(self):
             if assembly_type == AssemblyTypes.STANDARD:
                 # Attempt to move all advisories to QE
                 tasks = []
-                for impetus, advisory in impetus_advisories.items():
+                sorted_advisories = sorted(impetus_advisories.items())
```

Comment:
Is there a specific reason to sort them?

Reply:
yeah, it is to enable reporting when we get `results = asyncio.gather(*tasks, return_exceptions=True)`. The results are returned in order of the tasks given, so if you see the subsequent code, we log if there is an exception based on `sorted_advisories[i]`

Reply:
So IIUC, `tasks` will not contain the change microshift advisory state. You're then getting less results then sorted advisories. Doesn't this make indexing the advisories with the same `i` you're getting by looping over the results wrong?

Reply:
ah you're right, good catch